python/mlc_chat/nn/kv_cache.py

```diff
@@ -1049,6 +1062,9 @@ def _merge_state_inplace(
     VEC_SIZE = min(max(8 // v_dtype_bytes, head_dim // 32), 4)
     bdx = head_dim // VEC_SIZE
     bdy = num_heads
+    while bdx * bdy > target.max_num_threads and bdy > 1:
+        bdy //= 2
+    check_max_num_threads(target, bdx=bdx, bdy=bdy, bdz=1)
```

Comment:
Will this affect accuracy?

Reply:
Simply halving `bdy` might not be correct. May need to change the TIR as well.

Reply:
Updated with a gdy accordingly